pkg/api/universal/subtlecrypto_subtlecrypto.go

```diff
@@ -159,7 +163,8 @@ func (a *Universal) SubtleEncryptAlpha1(ctx context.Context, in *runtimev1pb.Sub
 		// We are not going to return the exact error from the component to the user, because an error that is too specific could allow for various side channel attacks (e.g. AES-CBC and padding oracle attacks)
 		// We will log the full error as a debug log, but only return a generic one to the user
 		a.logger.Debug(messages.ErrCryptoOperation.WithFormat(err))
-		err = messages.ErrCryptoOperation.WithFormat("failed to encrypt")
+		err = fmt.Errorf("failed to encrypt")
+		err = apierrors.CryptoProviderEncryptOperation(in.ComponentName, err)
```

Comment:
Nit: use [`errors.New`](https://pkg.go.dev/errors#New) if there's no format (like, no `%s`), which is faster (same below)

```suggestion
		err = apierrors.CryptoProviderEncryptOperation(in.ComponentName, errors.New("failed to encrypt"))
```